crates/wysiwyg/src/dom/dom_methods.rs

```diff
@@ -759,12 +759,16 @@ where
                 continue;
             }
             if let DomNode::Container(container) = self.lookup_node(&handle) {
-                if container.children().is_empty()
-                    && !(keep_empty_list_items && container.is_list_item())
-                {
-                    needs_removal = true;
-                } else if container.is_block_node()
-                    && container.only_contains_zwsp()
+                let children_are_empty = container.children().is_empty();
+
+                let is_list_item_and_keep_empty_list_items =
+                    keep_empty_list_items && container.is_list_item();
+
+                let is_block_with_only_zwsp =
+                    container.is_block_node() && container.only_contains_zwsp();
+
+                if children_are_empty && !is_list_item_and_keep_empty_list_items
+                    || is_block_with_only_zwsp
```

Comment:
Here the issue was two branches both resolving in the same way. I tried to tidy this up using variables for the booleans because when I put all of the logic checks together in the if check (as they were) I didn't think it looked that good.